### PR TITLE
BibTex enhancements and rate limiting fixes

### PR DESCRIPTION
**Pull request by Glignos**

### CODE REVIEW

**Comment by slint** [comment]
This also closes #1428. I think doing this for `dataset` should be fine also

File: zenodo/modules/records/serializers/bibtex.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def format(self):
             'poster': self._format_misc,
             'presentation': self._format_misc,
             'publication': self._format_publication,
-            'software': self._format_misc,
+            'software': self._format_software,
```
Comment (slint) [inline zenodo/modules/records/serializers/bibtex.py]: Add `dataset` as well to use the same logic

**Comment by katrinleinweber** [comment]
Thank you :-)

**Pull request merged**